homeassistant/components/watttime/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""The WattTime integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from aiowatttime import Client
+from aiowatttime.emissions import RealTimeEmissionsResponseType
+from aiowatttime.errors import WattTimeError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DATA_COORDINATOR, DOMAIN, LOGGER
+
+DEFAULT_UPDATE_INTERVAL = timedelta(minutes=5)
+
+PLATFORMS: list[str] = ["sensor"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up WattTime from a config entry."""
+    hass.data.setdefault(DOMAIN, {DATA_COORDINATOR: {}})
+
+    session = aiohttp_client.async_get_clientsession(hass)
+
+    try:
+        client = await Client.async_login(
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            session=session,
+            logger=LOGGER,
```

Comment:
Why should the client use our logger? Each package or module should preferably use its own logger, per standard library recommendations.

Reply:
`aiowatttime` supports it and I thought it'd be easier to debug with users if they only have to change `homeassistant.components.watttime`'s log level.

If it's really against standards, I can remove, but several other integrations (including some of my others) do this.

Reply:
It's standard to have a logger per package or module so that it's possible to resolve the log messages and apply different filters etc.